doc/advanced-configuration/listen.md

```diff
@@ -0,0 +1,738 @@
+
+#### `listen.http.handlers.mod_websockets.ping_rate`
+* **Syntax:** positive integer or the string `"none"`
+* **Default:** `"none"`
+* **Example:** `ping_rate = 10_000`
+
+The time between pings sent by server. By setting this option you enable server-side pinging.
+
+#### `listen.http.handlers.mod_websockets.max_stanza_size`
+* **Syntax:** positive integer or the string `"infinity"`
+* **Default:** `"infinity"`
+* **Example:** `max_stanza_size = 10_000`
+
+Maximum allowed incoming stanza size.
+**Warning:** this limit is checked **after** the input data parsing, so it does not apply to the input data size itself.
+
+#### `listen.http.handlers.mod_websockets.service`
+* **Syntax:** an array of `listen.service.*` options
+* **Default:** not set
+* **Example:**
+
+```toml
+[listen.http.handlers.mod_websockets.service]
+  access = "all"
+  shaper_rule = "fast"
+  password = "secret"
+```
+
+This subsection enables external component connections over WebSockets.
+See the [service](#xmpp-components-listenservice) listener section for details.
+
+### Handler types: REST API - Admin - `mongoose_api_admin`
+
+For more information about the API, see the [REST interface](../rest-api/Administration-backend.md) documentation.
+The following options are supported for this handler:
+
+#### `listen.http.handlers.mongoose_api_admin.username`
+* **Syntax:** string
+* **Default:** not set
+* **Example:** `username = "admin"`
+
+When set, enables authentication for the admin API, otherwise it is disabled. Requires setting `password`.
+
+#### `listen.http.handlers.mongoose_api_admin.password`
+* **Syntax:** string
+* **Default:** not set
+* **Example:** `password = "secret"`
+
+Required to enable authentication for the admin API.
+
+### Handler types: REST API - Client
+
+To enable the REST API for clients, several handlers need to be added:
+
+* `mongoose_client_api_*` - handles individual API endpoints. You can add and remove these to enable particular functionality.
+* `lasse_handler` - provides the [SSE handler](https://github.com/inaka/lasse) which is required for the client HTTP API, should not be changed.
+* `cowboy_*` - hosts the Swagger web-based documentation, should not be changed, but can be removed to disable the API docs.
+
+The recommended configuration is shown in [Example 3](#example-3-client-api) below.
+Please refer to [REST interface](../rest-api/Client-frontend.md) documentation for more information.
+
+### Handler types: Metrics API (obsolete) - `mongoose_api`
+
+REST API for accessing the internal MongooseIM metrics.
+Please refer to the [REST interface to metrics](../rest-api/Metrics-backend.md) page for more information.
+The following option is required:
+
+#### `listen.http.handlers.mongoose_api.modules`
+* **Syntax:** array of strings - Erlang modules
+* **Default:** not set, this is a mandatory option for this handler
+* **Example:** `modules = [mongoose_api_metrics]`
```

Comment:
If I understand correctly, this option is called `handlers` and not `modules` in our parser. Also, it should be:
```suggestion
* **Example:** `modules = ["mongoose_api_metrics"]`
```